src/Services/FileLoader/CommonMarkExtensions/ImageSizeParser.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+
+namespace App\Services\FileLoader\CommonMarkExtensions;
+
+
+use League\CommonMark\Inline\Element\Image;
+use League\CommonMark\Inline\Parser\InlineParserInterface;
+use League\CommonMark\InlineParserContext;
+
+class ImageSizeParser implements InlineParserInterface
+{
+
+    public function getCharacters(): array
+    {
+        return ["!"];
+    }
+
+    public function parse(InlineParserContext $inlineContext): bool
```

Comment:
Break this function into small reusable chunks will help maintaining the code.
Also a quick comment to explain whats happening here